src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogHttpClient.cs

```diff
@@ -36,123 +41,139 @@ public class DatadogHttpClient : IDatadogClient
         /// </summary>
         private const int MaxBackoff = 30;
 
-        public DatadogHttpClient(DatadogConfiguration config, LogFormatter formatter, string apiKey)
+        public DatadogHttpClient(DatadogConfiguration config, LogFormatter formatter, string apiKey, CancellationToken token)
         {
-            _config = config;
+            _cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token);
+            _cancellationToken = _cancellationTokenSource.Token;
             _client = new HttpClient();
+            _client.BaseAddress = new Uri(config.Url);
             _client.DefaultRequestHeaders.Add("DD-API-KEY", apiKey);
             _client.DefaultRequestHeaders.Add("DD-EVP-ORIGIN", "Serilog.Sinks.Datadog.Logs");
             _client.DefaultRequestHeaders.Add("DD-EVP-ORIGIN-VERSION", _version);
-            _url = $"{config.Url}/api/v2/logs";
             _formatter = formatter;
         }
 
-        public Task WriteAsync(IEnumerable<LogEvent> events)
+        /// <summary>
+        /// Writes the specified <paramref name="events"/> to Datadog logs route. 
+        /// </summary>
+        /// <param name="events">The events that will be sent.</param>
+        /// <param name="onException">The callback that is invoked when an exception occurs in this event but is handled gracefully.</param>
+        /// <remarks>
+        /// <para>This method uses a semaphore so that only one batch of events is ever on the wire at any given time.</para>
+        /// <para>This allows for a single allocated contiguous memory to be written to for serialization of events.</para>
```

Comment:
Thanks for introducing this. 
Would it be possible to put this behind a config flag? In scenarios where apps are sending a very high volume of logs someone may prioritize concurrent throughput over memory usage. The round trip latency of posting a payload to the intake and waiting for it to complete can impose a cap on how fast logs can be sent.

Reply:
Good catch. I can make that change.